heron/common/src/java/com/twitter/heron/common/basics/WakeableLooper.java

```diff
@@ -44,36 +44,62 @@
   private final PriorityQueue<TimerTask> timers;
 
   // The tasks would be invoked before exit
-  private final ArrayList<Runnable> exitTasks;
+  private final List<Runnable> exitTasks;
 
   // For selector since there is bug in selector.select(timeout): we could not
   // use a timeout > 10 * Integer.MAX_VALUE
   // So here we set Integer.MAX_VALUE as the infinite future
   // We will also multiple 1000*1000 to convert mill-seconds to nano-seconds
   private static final Duration INFINITE_FUTURE = Duration.ofMillis(Integer.MAX_VALUE);
   private volatile boolean exitLoop;
+  private boolean terminateAllTasksOnWakeup;
```

Comment:
Can we add comments for `terminateAllTasksOnWakeup ` and `terminateAllExitTasks `? Though I can understand them after I go through the code, it is still better to have more comments.